lib/ui/painting.dart

```diff
@@ -1839,6 +1839,14 @@ class Image {
     return stacks;
   }
 
+  /// Returns the approximate byte size of the allocation of this image.
+  ///
+  /// This is accurate for the storage format and mipmapping of the image.
+  int get approximateBytesUsed {
```

Comment:
I don't believe dlimage actually has storage format correct size estimations. We'd need to know about lossy or lossless texture compression as well.

Reply:
The current impeller implemenation is here: https://github.com/flutter/engine/blob/main/impeller/display_list/dl_image_impeller.cc#L88

While it correctly includes the format, it does not include mips. These should be added.

Additionally we should be able to check if texture compression is enabled. and include the compression ratios. approximately 0.5 for iOS backends.